### PR TITLE
chore(lint): enable usestdlibvars for HTTP/time constants + cleanup sweep

### LINKED ISSUE #1386: chore(lint): enable usestdlibvars for HTTP/time constants + cleanup sweep
## Goal
Enable `usestdlibvars` in `.golangci.yml`. Forces `http.StatusOK` etc. instead of literal status codes -- keeps handler responses lexically aligned with OpenAPI status enums and helps `TestOpenAPIConsistency`'s string-match heuristics get more reliable when handlers don't use raw integers.

## Implementation
1. Add `usestdlibvars` to `linters.enable`.
2. Run cleanup sweep on `internal/api/handlers_*.go`.
3. Bundle into one PR.

## Acceptance criteria
- [ ] Linter enabled in `.golangci.yml`.
- [ ] `grep -E '\b(200|400|404|500)\b' internal/api/handlers_*.go` returns no raw status codes (only documented exceptions).
- [ ] Lint baseline clean.

## Charter
Direct M49 fit -- pairs with #1094 (kin-openapi runtime validation) by reducing raw-integer drift between handlers and spec. Plan issue F3.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Enables `usestdlibvars` in `.golangci.yml` with all standard checks (http-method, http-status-code, time-layout, crypto-hash, etc.).
- Sweep totals: 49 findings across 18 source files.
  - **http-method** (35): `"GET"` -> `http.MethodGet`, `"POST"` -> `http.MethodPost`, `"DELETE"` -> `http.MethodDelete` -- mostly in test files using `httptest.NewRequest` / `http.NewRequestWithContext`.
  - **time-layout** (14): `"2006-01-02T15:04:05Z07:00"` -> `time.RFC3339`, `"2006-01-02"` -> `time.DateOnly`, `"2006-01-02 15:04:05"` -> `time.DateTime`. These constants are byte-equivalent to their pre-change literals (verified at compile time); zero runtime delta.
  - **http-status-code** (0): codebase already uses `http.Status*` constants in every `WriteHeader` / `http.Error` call.
  - **other checks** (0): no hits for crypto-hash, time-weekday/month, sql-isolation-level, etc.
- Zero `//nolint:usestdlibvars` directives needed.
- Acceptance-criteria grep `grep -E '\b(200|400|404|500)\b' internal/api/handlers_*.go` returns zero matches against actual HTTP status codes (residual matches are Tailwind CSS classes, comments, and unrelated integer constants like page sizes).
- Includes 5 targeted tests added in follow-up commits to cover the time-format paths the patch-coverage gate flagged. The renames are byte-equivalent so the tests are happy-path-only -- they exercise the converted code without expanding lint-sweep scope into broader coverage uplift work (M49 W5).
- Third of 5 sequential M49 W1 PRs (F1 exhaustive merged 32c8c353; F2 errorlint merged e6a2fe44; this is F3; F4 musttag / F5 gate-runs-lint to follow).

## Test plan

- [x] `golangci-lint run ./...` clean (usestdlibvars enabled, 0 findings)
- [x] `make build` clean
- [x] `go test ./...` all packages pass; new tests cover `parseHistoryTimestamp`, `parseNullableTime`, `toConnectionResponse`, `handleReportHealthHistory` (RFC3339 + DateOnly paths)
- [x] `bash scripts/pre-push-gate.sh` clean (race detector + OpenAPI + 100% patch coverage)

Closes #1386

## Summary by CodeRabbit

* **Refactor**
  * Standardized timestamp parsing and formatting across the app and APIs using consistent standard time formats (RFC3339, DateTime, DateOnly).
  * Replaced HTTP method string literals with typed constants in tests for consistency.

* **Tests**
  * Added and updated tests for timestamp parsing, connection timestamp outputs, and health-history date-range behavior.

* **Chores**
  * Strengthened lint configuration with additional standard-library variable checks.

### CODE REVIEW

File: internal/api/handlers_connection.go [hunk 1]
```diff
@@ -46,16 +46,16 @@ func toConnectionResponse(c connection.Connection) connectionResponse {
 		Enabled:               c.Enabled,
 		Status:                c.Status,
 		StatusMessage:         c.StatusMessage,
-		CreatedAt:             c.CreatedAt.Format("2006-01-02T15:04:05Z07:00"),
-		UpdatedAt:             c.UpdatedAt.Format("2006-01-02T15:04:05Z07:00"),
+		CreatedAt:             c.CreatedAt.Format(time.RFC3339),
+		UpdatedAt:             c.UpdatedAt.Format(time.RFC3339),
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_connection.go]: Fixed in c59e0dda. Added `.UTC()` before `.Format(time.RFC3339)` on all three timestamp fields (CreatedAt, UpdatedAt, LastCheckedAt) to match the OpenAPI UTC RFC3339 contract and align with the codebase pattern. Also strengthened TestToConnectionResponse_RFC3339Fields to construct inputs in America/Los_Angeles so the .UTC() call is genuinely exercised and asserts the formatted output ends with "Z".

File: internal/api/handlers_report_test.go [hunk 2]
```diff
@@ -730,6 +730,59 @@ func TestHandleViolationTrend_UpperBoundClamped(t *testing.T) {
...
+
+	var resp map[string][]rule.HealthSnapshot
+	if err := json.NewDecoder(w.Body).Decode(&resp); err != nil {
+		t.Fatalf("decoding response: %v", err)
+	}
+
+	if _, ok := resp["history"]; !ok {
+		t.Error("response missing 'history' key")
+	}
+}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_report_test.go:797]: Fixed in c59e0dda. Both tests now seed four health_history snapshots (one before, two within, one after the queried range) and assert exactly the in-range set is returned with the expected scores in ascending order. Added a `seedHealthSnapshot` helper that inserts directly to `health_history` with explicit recorded_at since `Service.RecordHealthSnapshot` uses `time.Now()` and a 5-minute throttle.

**Comment by sydlexius** [comment]
@coderabbitai resolve

File: internal/api/handlers_report_test.go [hunk 3]
```diff
@@ -730,6 +730,113 @@ func TestHandleViolationTrend_UpperBoundClamped(t *testing.T) {
...
+func TestHandleReportHealthHistory_DateOnlyParams(t *testing.T) {
+	t.Parallel()
+	r, _ := testRouter(t)
+
+	// Seeded snapshots: 2 outside range (50, 80), 2 inside (60, 70).
+	// Query window: 2026-01-01 to 2026-12-31 (date-only -> midnight UTC bounds).
+	// In-range snapshots are placed well clear of the boundary instants so the
+	// date-only and RFC3339 tests can share the same data without boundary
+	// edge-case differences (date-only "to" is 2026-12-31T00:00:00Z while
+	// RFC3339 "to" is end-of-day).
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_report_test.go]: Fixed in e1c3ff3. Made date-only `to` inclusive of end-of-day in the handler (`t.Add(24*time.Hour - time.Second)` on the DateOnly parse branch) and added the suggested boundary assertions (`2026-12-31T23:59:59Z` included, `2027-01-01T00:00:00Z` excluded) to both `TestHandleReportHealthHistory_DateOnlyParams` and the RFC3339 variant so both code paths enforce the same inclusive-day contract.
